### ISSUE TITLE
Rant: Please make Portfolio Performance friendly for geeks

### ISSUE THREAD

**Issue by pfalcon** [portfolio-performance/portfolio#3303]
Portfolio Performance is incredibly powerful tool, it is also incredibly user friendly, as much as an open-source software can be end user friendly. But there're limits on how much software can be user-friendly, while being resilient to errors and similar. And as long as software depends on 3rd-party world (network services and APIs), it's in the constant state of breakage. https://github.com/buchen/portfolio/issues/3275, https://github.com/buchen/portfolio/issues/3302 are just couple of examples seen by a novice user. In both cases, discovering that something was wrong (beyond "it doesn't work!!11"), and what exactly was wrong, was pretty hard. Definitely harder than it could have been. So, can we please:

1. Have console logging enabled by default, and all the errors, etc. be logged there with enough context (like URLs accessed).
2. Actually more then that, it would be nice that errors lead to nice error dialogs in user's face with beautiful hundred-lines stacktraces. That way the errors won't go unnoticed. (Balancing that of course not easy, a user shouldn't get 100 dialogs in row, so that's not replacement for console log).

I.e. let those users who know how to deal with software errors, be able to deal with them without extraordinary effort on just getting access to them. And even users who don't know how to deal with them, at least would be able to report them with objective details.

Thanks for listening.

**Comment by Nirus2000** [comment]
Hmm... if you don't quite like each other, then please stick to the technical, factual level.
This is getting childish... 
It is a 10 years old open source project and everyone can contribute. The time spent is more towards functionality and not making an error log more accessible or easier to read for beginners. 
I also see this point not as a BUG but as a FEATURE.

**Comment by chirlu** [comment]
I don’t know @pfalcon from before. What I don’t like is this behaviour (in which he is not alone) of “I don’t immediately find some feature – let’s not ask nicely on the forums whether I might have overlooked it or whether there is some other way to achieve the desired result, let’s immediately file a bug report with some condescending remarks about how bad everything is”.

**Comment by Nirus2000** [comment]
I'm right there with you that this is out of place and lacks netiquette.

**Comment by pfalcon** [comment]
> “I don’t immediately find some feature
> let’s immediately file a bug report 

Trust me, I do my homework on looking for things, including in the source code, and post only then (in the worst case, just a reminder for myself that I have done some work - but of course, always with the hope that someone else may find it useful).

Btw, this was filed as a "feature request", not a "bug report" (but generally is an open-ended discussion, though some points are actionable in sustainable (not immediate) manner IMHO).

> with some condescending remarks

My reports are definitely terse and might come as stern - the result of lack of time to write it in more "oiled-up" manner, many people sadly choose between just writing something or not write all. Sorry if that instead sounded condescending. At least I didn't call anybody annoying and didn't suggest them to shut up, as some participants here did in other tickets.

> about how bad everything is

D'oh. "Portfolio Performance is incredibly powerful tool, it is also incredibly user friendly". Talk about difference in perception of what's "bad" and what's "condescending".

---

Anyway, I posted this as a kind of open-ended discussion, a kind of reply to survey link to which is available in the UI of the recent PP versions. I don't expect any immediate actions, just discussing opportunity of adjusting the course of PP, taking nautical metaphor, 5-10 degrees towards being more hacker-friendly (again, it's end-user friendliness is very great trait which shouldn't be lost, just new "doors" to be opened up). I'd like to contribute towards that direction - as long as I would be able/interested to contribute at all, of which I'm at early stages of. Of course, I wouldn't be able to contribute towards that without discussing it well beforehand and others being positive of that idea.

**Comment by pfalcon** [comment]
> Ah, those great actionable issues.

As mentioned above, this is more like "strategic outlook" ticket, but from it, quite specific, actionable issues can be entailed: https://github.com/buchen/portfolio/issues/3417

**Comment by Nirus2000** [comment]
@pfalcon
Yes, we read an ISSUE, but so far not much came from your side on this topic. Wishing for changes, but not pushing anything forward.... think about it...
I would ask you to refrain from such discussions in the ISSUE and not to deposit in the "pull request of another contributer or new contributer"....
